### PR TITLE
copier: add Remove()

### PR DESCRIPTION
**Pull request by nalind**
#### What type of PR is this?

/kind other

#### What this PR does / why we need it:

Add copier.Remove().

#### How to verify it

New unit tests!

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```
None
```

### CODE REVIEW

**Review by ashley-cui — COMMENTED** [review]
LGTM, thanks!

**Comment by ashley-cui** [comment]
@containers/build-maintainers PTAL

File: copier/copier.go [hunk 1]
```diff
@@ -1785,3 +1831,22 @@ func copierHandlerMkdir(req request, idMappings *idtools.IDMappings) (*response,
...
+	if err != nil {
+		return errorResponse("copier: remove: error resolving %q: %v", req.Directory, err)
+	}
+	if req.RemoveOptions.All {
+		err = os.RemoveAll(resolvedTarget)
+	} else {
+		err = os.Remove(resolvedTarget)
+	}
+	if err != nil {
+		return errorResponse("copier: remove: error removing %q: %v", req.Directory, err)
```
Comment (rhatdan) [inline copier/copier.go:1849]: This is a stutter.  Drop the `removing %q`

File: copier/copier.go [hunk 2]
```diff
@@ -1785,3 +1831,22 @@ func copierHandlerMkdir(req request, idMappings *idtools.IDMappings) (*response,
...
 	return &response{Error: "", Mkdir: mkdirResponse{}}, nil, nil
 }
+
+func copierHandlerRemove(req request) *response {
+	errorResponse := func(fmtspec string, args ...interface{}) *response {
+		return &response{Error: fmt.Sprintf(fmtspec, args...), Remove: removeResponse{}}
+	}
+	resolvedTarget, err := resolvePath(req.Root, req.Directory, false, nil)
+	if err != nil {
+		return errorResponse("copier: remove: error resolving %q: %v", req.Directory, err)
```
Comment (rhatdan) [inline copier/copier.go:1841]: This is another stutter.  resolvePath reports the directory.

File: copier/copier.go (hunk 1 above)
Comment (nalind) [reply copier/copier.go:1849]: Changing the format string to `copier: remove %q: %v`.

File: copier/copier.go (hunk 2 above)
Comment (nalind) [reply copier/copier.go:1841]: Replacing the format string with `copier: remove: %v`.

File: copier/copier.go [hunk 3]
```diff
@@ -1785,3 +1831,22 @@ func copierHandlerMkdir(req request, idMappings *idtools.IDMappings) (*response,
...
+	if err != nil {
+		return errorResponse("copier: remove: %v", err)
+	}
+	if req.RemoveOptions.All {
+		err = os.RemoveAll(resolvedTarget)
+	} else {
+		err = os.Remove(resolvedTarget)
+	}
+	if err != nil {
+		return errorResponse("copier: remove %q: %v", req.Directory, err)
```
Comment (rhatdan) [inline copier/copier.go:1849]: Still a bit of a stutter, but I guess req.Directory != resolvedTarget
/lgtm
/hold

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Pull request merged**